diff --git a/frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx b/frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx
index 3d8de356969..41a41d133b4 100644
--- a/frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx
+++ b/frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx
@@ -1,4 +1,4 @@
-import { useEffect, useState } from "react";
+import { useState } from "react";
 
 import Question from "metabase-lib/v1/Question";
 
@@ -20,12 +20,6 @@ export function ObjectDetail({
 }: ObjectDetailProps) {
   const [currentObjectIndex, setCurrentObjectIndex] = useState(0);
 
-  useEffect(() => {
-    if (data.rows.length <= currentObjectIndex) {
-      setCurrentObjectIndex(0);
-    }
-  }, [data.rows, currentObjectIndex]);
-
   const hasPagination = data?.rows?.length > 1;
   const resolvedQuestion =
     question ??
