diff --git a/src/metabase/lib/drill_thru/pivot.cljc b/src/metabase/lib/drill_thru/pivot.cljc
index 6e141b65fef..07eeee9f2ff 100644
--- a/src/metabase/lib/drill_thru/pivot.cljc
+++ b/src/metabase/lib/drill_thru/pivot.cljc
@@ -139,6 +139,8 @@
                (some? value)
                (not dimensions-are-untraceable?)
                (lib.underlying/aggregation-sourced? query column)
+               ;; TODO fix this drill thru and remove this check (metabase#52236)
+               (not (lib.underlying/strictly-underlying-aggregation? query column))
                (-> (lib.aggregation/aggregations query stage-number) count pos?))
       (let [breakout-pivot-types (permitted-pivot-types query stage-number)
             pivots               (into {} (for [pivot-type breakout-pivot-types
