diff --git a/frontend/src/metabase/query_builder/components/view/ViewHeader/components/QuestionDataSource/SourceQuestionBreadcrumbs.tsx b/frontend/src/metabase/query_builder/components/view/ViewHeader/components/QuestionDataSource/SourceQuestionBreadcrumbs.tsx
index fb204a46913..c593a04a9a9 100644
--- a/frontend/src/metabase/query_builder/components/view/ViewHeader/components/QuestionDataSource/SourceQuestionBreadcrumbs.tsx
+++ b/frontend/src/metabase/query_builder/components/view/ViewHeader/components/QuestionDataSource/SourceQuestionBreadcrumbs.tsx
@@ -7,7 +7,6 @@ import * as Lib from "metabase-lib";
 import Question from "metabase-lib/v1/Question";
 import { getQuestionIdFromVirtualTableId } from "metabase-lib/v1/metadata/utils/saved-questions";
 
-import { DataSourceCrumbs } from "./DataSourceCrumbs";
 import { SourceModelBreadcrumbs } from "./SourceModelBreadcrumbs";
 
 interface Props {
@@ -37,15 +36,11 @@ export function SourceQuestionBreadcrumbs({
     return null;
   }
 
-  if (sourceQuestion.type() === "model" || sourceQuestion.type() === "metric") {
-    return (
-      <SourceModelBreadcrumbs
-        question={sourceQuestion}
-        variant={variant}
-        {...props}
-      />
-    );
-  }
-
-  return <DataSourceCrumbs question={question} variant={variant} {...props} />;
+  return (
+    <SourceModelBreadcrumbs
+      question={sourceQuestion}
+      variant={variant}
+      {...props}
+    />
+  );
 }
