diff --git a/node_modules/eslint-plugin-i18next/lib/helper/matchPatterns.js b/node_modules/eslint-plugin-i18next/lib/helper/matchPatterns.js
--- a/node_modules/eslint-plugin-i18next/lib/helper/matchPatterns.js
+++ b/node_modules/eslint-plugin-i18next/lib/helper/matchPatterns.js
@@ -7,9 +7,8 @@
   if (handler) {
     return handler(text);
   }
-  handler = str => {
-    return patterns.map(generateFullMatchRegExp).some(item => item.test(str));
-  };
+  const expressions = patterns.map(generateFullMatchRegExp);
+  handler = str => expressions.some(item => item.test(str));
   cache.set(patterns, handler);
   return handler(text);
 };
diff --git a/node_modules/eslint-plugin-i18next/lib/rules/no-literal-string.js b/node_modules/eslint-plugin-i18next/lib/rules/no-literal-string.js
--- a/node_modules/eslint-plugin-i18next/lib/rules/no-literal-string.js
+++ b/node_modules/eslint-plugin-i18next/lib/rules/no-literal-string.js
@@ -15,7 +15,7 @@
 function isValidFunctionCall(context, options, { callee }) {
   if (callee.type === 'Import') return true;
 
-  const sourceText = context.getSourceCode().getText(callee);
+  const sourceText = context.sourceCode.getText(callee);
 
   return shouldSkip(options.callees, sourceText);
 }
@@ -109,7 +109,7 @@
     function report(node) {
       context.report({
         node,
-        message: `${message}: ${context.getSourceCode().getText(node.parent)}`,
+        message: `${message}: ${context.sourceCode.getText(node.parent)}`,
       });
     }
 
@@ -152,7 +152,7 @@
     function filterOutJSX(node) {
       if (onlyValidateJSX) {
         const isInsideJSX = (
-          context.getAncestors || context.sourceCode.getAncestors
+          context.sourceCode.getAncestors
         )(node).some(item => ['JSXElement', 'JSXFragment'].includes(item.type));
 
         if (!isInsideJSX) return true;
@@ -375,7 +375,7 @@
 
         if (onlyValidateVueTemplate) {
           const parents = (
-            context.getAncestors || context.sourceCode.getAncestors
+            context.sourceCode.getAncestors
           )(node);
           if (
             parents.length &&
