diff --git a/node_modules/password-generator/lib/password-generator.js b/node_modules/password-generator/lib/password-generator.js
index 4048bc1..ab44c36 100644
--- a/node_modules/password-generator/lib/password-generator.js
+++ b/node_modules/password-generator/lib/password-generator.js
@@ -91,13 +91,16 @@


   getRandomValues = function (buf) {
+    root = globalThis;
     if (root.crypto && root.crypto.getRandomValues) {
       root.crypto.getRandomValues(buf);
     } else if (typeof root.msCrypto === "object" && typeof root.msCrypto.getRandomValues === 'function') {
       root.msCrypto.getRandomValues(buf);
+    // require('crypto') fails due to bundler's static analysis
+    // both node and browsers have getRandomValues nowadays
-    } else if (module.exports === password && typeof require !== "undefined") {
-      var bytes = require("crypto").randomBytes(buf.length);
-      buf.set(bytes);
     } else {
       throw new Error("No secure random number generator available.");
     }
