diff --git a/lib/password-generator.js b/lib/password-generator.js
index 4048bc1..e5d478b 100644
--- a/lib/password-generator.js
+++ b/lib/password-generator.js
@@ -91,13 +91,15 @@
 
 
   getRandomValues = function (buf) {
+    // override root to be able to run in both node and browser env
+    root = globalThis;
     if (root.crypto && root.crypto.getRandomValues) {
       root.crypto.getRandomValues(buf);
     } else if (typeof root.msCrypto === "object" && typeof root.msCrypto.getRandomValues === 'function') {
       root.msCrypto.getRandomValues(buf);
-    } else if (module.exports === password && typeof require !== "undefined") {
-      var bytes = require("crypto").randomBytes(buf.length);
-      buf.set(bytes);
+      // require('crypto') fails due to bundler's static analysis
+      // both node and browsers have getRandomValues nowadays,
+      // so require is removed
     } else {
       throw new Error("No secure random number generator available.");
     }
