diff --git a/node_modules/password-generator/lib/password-generator.js b/node_modules/password-generator/lib/password-generator.js
index 4048bc1..25faa33 100644
--- a/node_modules/password-generator/lib/password-generator.js
+++ b/node_modules/password-generator/lib/password-generator.js
@@ -91,10 +91,10 @@
 
 
   getRandomValues = function (buf) {
-    if (root.crypto && root.crypto.getRandomValues) {
-      root.crypto.getRandomValues(buf);
-    } else if (typeof root.msCrypto === "object" && typeof root.msCrypto.getRandomValues === 'function') {
-      root.msCrypto.getRandomValues(buf);
+    if (globalThis.crypto && globalThis.crypto.getRandomValues) {
+      globalThis.crypto.getRandomValues(buf);
+    } else if (typeof globalThis.msCrypto === "object" && typeof globalThis.msCrypto.getRandomValues === 'function') {
+      globalThis.msCrypto.getRandomValues(buf);
     } else if (module.exports === password && typeof require !== "undefined") {
       var bytes = require("crypto").randomBytes(buf.length);
       buf.set(bytes);
