diff --git a/node_modules/zrender/lib/svg/patch.js b/node_modules/zrender/lib/svg/patch.js
index 39fa751..f11447a 100644
--- a/node_modules/zrender/lib/svg/patch.js
+++ b/node_modules/zrender/lib/svg/patch.js
@@ -95,7 +95,10 @@ export function updateAttrs(oldVnode, vnode) {
                 elm.removeAttribute(key);
             }
             else {
-                if (key.charCodeAt(0) !== xChar) {
+                if (key === 'style') {
+                    elm.style.cssText = cur;
+                }
+                else if (key.charCodeAt(0) !== xChar) {
                     elm.setAttribute(key, cur);
                 }
                 else if (key === 'xmlns:xlink' || key === 'xmlns') {
