diff --git a/frontend/src/metabase/visualizations/visualizations/Map/definition.tsx b/frontend/src/metabase/visualizations/visualizations/Map/definition.tsx
index 55ad9834606..afcb58fcc36 100644
--- a/frontend/src/metabase/visualizations/visualizations/Map/definition.tsx
+++ b/frontend/src/metabase/visualizations/visualizations/Map/definition.tsx
@@ -131,8 +131,8 @@ export const MAP_VIZ_DEFINITION: VisualizationDefinition = {
           : vizSettings["map.type"] === "grid"
             ? "grid"
             : data.rows.length >= 1000
-              ? "tiles"
-              : "markers",
+              ? "markers"
+              : "tiles",
       getHidden: (_series, vizSettings) =>
         !isPinMapType(vizSettings["map.type"]),
     },
