diff --git a/frontend/src/metabase/querying/components/QueryVisualization/QueryVisualization.tsx b/frontend/src/metabase/querying/components/QueryVisualization/QueryVisualization.tsx
index 10024917099..b5f9a943f1d 100644
--- a/frontend/src/metabase/querying/components/QueryVisualization/QueryVisualization.tsx
+++ b/frontend/src/metabase/querying/components/QueryVisualization/QueryVisualization.tsx
@@ -12,7 +12,6 @@ import { useSelector } from "metabase/redux";
 import { getWhiteLabeledLoadingMessageFactory } from "metabase/selectors/whitelabel";
 import { Box, Flex, Stack, Text, Title } from "metabase/ui";
 import { isMac } from "metabase/utils/browser";
-import { SERVER_ERROR_TYPES } from "metabase/utils/errors";
 import * as Lib from "metabase-lib";
 import { HARD_ROW_LIMIT } from "metabase-lib/v1/queries/utils";
 
@@ -46,9 +45,7 @@ export function QueryVisualization(props: QueryVisualizationProps) {
     isRunnable &&
     !isRunning &&
     !isNativeEditorOpen &&
-    (result?.error == null ||
-      isDirtyStateShownForError ||
-      result?.error_type === SERVER_ERROR_TYPES.missingRequiredParameter);
+    (result?.error == null || isDirtyStateShownForError);
 
   return (
     <Box pos="relative" h="100%" className={cx(className, S.root)}>
