diff --git a/frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx b/frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx
index 8027aaafa60..9ffd0e196b7 100644
--- a/frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx
+++ b/frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx
@@ -139,6 +139,7 @@ export class ClickActionsPopover extends Component<
             this.close();
           }
         }}
+        trapFocus
         position="bottom-start"
         offset={8}
         popoverContentTestId="click-actions-popover"
