diff --git a/frontend/src/metabase-lib/v1/parameters/utils/filters.ts b/frontend/src/metabase-lib/v1/parameters/utils/filters.ts
index e7ed076e74a..b8bd59618d7 100644
--- a/frontend/src/metabase-lib/v1/parameters/utils/filters.ts
+++ b/frontend/src/metabase-lib/v1/parameters/utils/filters.ts
@@ -21,7 +21,6 @@ import {
   isFK,
   isNumeric,
   isPK,
-  isString,
   isStringLike,
 } from "../../types/utils/isa";
 
@@ -87,7 +86,7 @@ export function fieldFilterForParameter(
 ): (field: ParameterFilterableField) => boolean {
   return (field) =>
     isParameterCompatibleWithColumn(parameter, {
-      isString: isString(field) || isStringLike(field),
+      isString: isStringLike(field),
       isNumeric: isNumeric(field),
       isBoolean: isBoolean(field),
       isTemporal: isDate(field),
