diff --git a/frontend/src/metabase/parameters/utils/mapping-options.ts b/frontend/src/metabase/parameters/utils/mapping-options.ts
index b24fbe52fa7..55030c8300c 100644
--- a/frontend/src/metabase/parameters/utils/mapping-options.ts
+++ b/frontend/src/metabase/parameters/utils/mapping-options.ts
@@ -302,48 +302,37 @@ export function getMappingOptionByTarget(
   }
 
   const { query, columns } = getParameterColumns(question, parameter);
-  const stageCount = Lib.stageCount(query);
-  const lastStageIndex = stageCount - 1;
-  const stageIndex = getStageIndexFromTarget(target) ?? lastStageIndex;
-  if (stageIndex >= stageCount) {
-    return;
-  }
-
-  const stageColumns = columns
-    .filter((column) => column.stageIndex === stageIndex)
-    .map(({ column }) => column);
-  const stageMappingOptions = mappingOptions.filter(
-    ({ target }) => getStageIndexFromTarget(target) === stageIndex,
-  );
-
+  const stageIndexes = _.uniq(columns.map(({ stageIndex }) => stageIndex));
   const normalizedTarget = normalize(target);
   const fieldRef = normalizedTarget[1];
-  const [columnByTargetIndex] = Lib.findColumnIndexesFromLegacyRefs(
-    query,
-    stageIndex,
-    stageColumns,
-    [fieldRef],
-  );
 
-  if (columnByTargetIndex !== -1) {
-    const mappingColumnIndexes = Lib.findColumnIndexesFromLegacyRefs(
+  for (const stageIndex of stageIndexes) {
+    const stageColumns = columns
+      .filter((column) => column.stageIndex === stageIndex)
+      .map(({ column }) => column);
+
+    const [columnByTargetIndex] = Lib.findColumnIndexesFromLegacyRefs(
       query,
       stageIndex,
       stageColumns,
-      stageMappingOptions.map(({ target }) => target[1] as DimensionReference),
+      [fieldRef],
     );
 
-    const mappingIndex = mappingColumnIndexes.indexOf(columnByTargetIndex);
-    if (mappingIndex >= 0) {
-      return stageMappingOptions[mappingIndex];
+    if (columnByTargetIndex !== -1) {
+      const mappingColumnIndexes = Lib.findColumnIndexesFromLegacyRefs(
+        query,
+        stageIndex,
+        stageColumns,
+        mappingOptions.map(({ target }) => target[1] as DimensionReference),
+      );
+
+      const mappingIndex = mappingColumnIndexes.indexOf(columnByTargetIndex);
+
+      if (mappingIndex >= 0) {
+        return mappingOptions[mappingIndex];
+      }
     }
   }
 
   return undefined;
 }
-
-function getStageIndexFromTarget(target: ParameterTarget): number | undefined {
-  if (isStructuredDimensionTarget(target)) {
-    return target[2]?.["stage-number"];
-  }
-}
