diff --git a/frontend/src/metabase/query_builder/actions/url.ts b/frontend/src/metabase/query_builder/actions/url.ts
index 21140533d8e..ad443540b09 100644
--- a/frontend/src/metabase/query_builder/actions/url.ts
+++ b/frontend/src/metabase/query_builder/actions/url.ts
@@ -11,7 +11,6 @@ import type Question from "metabase-lib/v1/Question";
 import { isAdHocModelOrMetricQuestion } from "metabase-lib/v1/metadata/utils/models";
 
 import {
-  getCard,
   getDatasetEditorTab,
   getOriginalQuestion,
   getQueryBuilderMode,
@@ -81,9 +80,7 @@ export const updateUrl = createThunkAction(
         datasetEditorTab = getDatasetEditorTab(getState());
       }
 
-      const card = isAdHocModelOrMetric
-        ? getCard(getState())!
-        : question.card();
+      const card = question.card();
       const newState = {
         card,
         cardId: question.id(),
