diff --git a/frontend/src/metabase/query_builder/reducers.ts b/frontend/src/metabase/query_builder/reducers.ts
index 1b886b42fb7..4ef5a6a1f0a 100644
--- a/frontend/src/metabase/query_builder/reducers.ts
+++ b/frontend/src/metabase/query_builder/reducers.ts
@@ -307,7 +307,6 @@ export const uiControls = createReducer<QueryBuilderUIControls>(
       .addCase(OPEN_QUESTION_INFO, (state) =>
         setUIControls(state, {
           ...UI_CONTROLS_SIDEBAR_DEFAULTS,
-          ...CLOSED_NATIVE_EDITOR_SIDEBARS,
           isShowingQuestionInfoSidebar: true,
         }),
       )
@@ -325,12 +324,16 @@ export const uiControls = createReducer<QueryBuilderUIControls>(
         ...state,
         isShowingQuestionSettingsSidebar: false,
       }))
-      .addCase(OPEN_TIMELINES, (state) => ({
-        ...state,
-        ...UI_CONTROLS_SIDEBAR_DEFAULTS,
-        ...CLOSED_NATIVE_EDITOR_SIDEBARS,
-        isShowingTimelineSidebar: true,
-      }))
+      .addCase<string, { type: string; payload: number[] | undefined }>(
+        OPEN_TIMELINES,
+        (state, action) => ({
+          ...state,
+          ...UI_CONTROLS_SIDEBAR_DEFAULTS,
+          ...CLOSED_NATIVE_EDITOR_SIDEBARS,
+          isShowingTimelineSidebar: true,
+          focusedTimelineEventIds: action.payload ?? null,
+        }),
+      )
       .addCase(CLOSE_TIMELINES, (state) => ({
         ...state,
         ...UI_CONTROLS_SIDEBAR_DEFAULTS,
