diff --git a/frontend/src/metabase/querying/expressions/tokenizer/tokens.ts b/frontend/src/metabase/querying/expressions/tokenizer/tokens.ts
index 538690c308a..c7fcaf069fc 100644
--- a/frontend/src/metabase/querying/expressions/tokenizer/tokens.ts
+++ b/frontend/src/metabase/querying/expressions/tokenizer/tokens.ts
@@ -23,7 +23,7 @@ const EOF = -1;
 //   [Field Name
 //   ------
 //
-const FIELD_PUNCTUATORS = new Set([char(","), char(" "), char("("), char(")")]);
+const FIELD_PUNCTUATORS = new Set([char(" "), char("("), char(")")]);
 
 /**
  * Reference (or bracket identifiers) like `[User Id]` are parsed differently
