diff --git a/src/metabase/embedding_rest/api/embed.clj b/src/metabase/embedding_rest/api/embed.clj
index 13c2cb2c341..655e1eee7bd 100644
--- a/src/metabase/embedding_rest/api/embed.clj
+++ b/src/metabase/embedding_rest/api/embed.clj
@@ -414,12 +414,12 @@
         card-id (api.embed.common/unsigned-token->card-id unsigned)
         card (api/check-404 (embedding-rest.db/card card-id))]
     (api.embed.common/check-embedding-enabled-for-card card)
-    (request/as-admin
-      (api.embed.common/process-tiles-query-for-card
-       card
+    (api.embed.common/process-tiles-query-for-card
+     card
+     (request/as-admin
        (api.embed.common/tile-parameters-for-card
-        card (embedding.jwt/get-in-unsigned-token-or-throw unsigned [:params]) parameters)
-       zoom x y latField lonField))))
+        card (embedding.jwt/get-in-unsigned-token-or-throw unsigned [:params]) parameters))
+     zoom x y latField lonField)))
 
 ;; TODO (Cam 2025-11-25) please add a response schema to this API endpoint, it makes it easier for our customers to
 ;; use our API + we will need it when we make auto-TypeScript-signature generation happen
