diff --git a/frontend/src/metabase/query_builder/components/DatasetEditor/DatasetQueryEditor.tsx b/frontend/src/metabase/query_builder/components/DatasetEditor/DatasetQueryEditor.tsx
index 824c20af637..e3c2b8ab1cd 100644
--- a/frontend/src/metabase/query_builder/components/DatasetEditor/DatasetQueryEditor.tsx
+++ b/frontend/src/metabase/query_builder/components/DatasetEditor/DatasetQueryEditor.tsx
@@ -114,4 +114,9 @@ function DatasetQueryEditorInner({
   );
 }
 
-export const DatasetQueryEditor = memo(DatasetQueryEditorInner);
+export const DatasetQueryEditor = memo(
+  DatasetQueryEditorInner,
+  // should prevent the editor from re-rendering in "metadata" mode
+  // when it's completely covered with the results table
+  (prevProps, nextProps) => prevProps.height === 0 && nextProps.height === 0,
+);
