diff --git a/frontend/src/metabase/metadata/components/FkTargetPicker/FkTargetPicker.tsx b/frontend/src/metabase/metadata/components/FkTargetPicker/FkTargetPicker.tsx
index 0f7b833b24a..c04b6a93111 100644
--- a/frontend/src/metabase/metadata/components/FkTargetPicker/FkTargetPicker.tsx
+++ b/frontend/src/metabase/metadata/components/FkTargetPicker/FkTargetPicker.tsx
@@ -76,9 +76,8 @@ export const FkTargetPicker = ({
           }
 
           return (
-            field.display_name.includes(query) ||
-            field.table?.display_name.includes(query) ||
-            field.table?.schema_name?.includes(query)
+            option.label.toLowerCase().includes(query) ||
+            field.description?.toLowerCase().includes(query)
           );
         });
       }}
