diff --git a/frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx b/frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx
index b0a19d5d4e3..e6b20fba260 100644
--- a/frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx
+++ b/frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx
@@ -2,9 +2,8 @@ import type { SyntheticEvent, TransitionEventHandler } from "react";
 import { forwardRef, useEffect, useState } from "react";
 import type { ResizableBoxProps, ResizeCallbackData } from "react-resizable";
 import { ResizableBox } from "react-resizable";
-import { useWindowSize } from "react-use";
+import { useMedia, useWindowSize } from "react-use";
 
-import { useIsSmallScreen } from "metabase/common/hooks/use-is-small-screen";
 import { setNotebookNativePreviewSidebarWidth } from "metabase/query_builder/actions";
 import { getUiControls } from "metabase/query_builder/selectors";
 import {
@@ -80,7 +79,7 @@ export const NotebookContainer = ({
     dispatch(setNotebookNativePreviewSidebarWidth(width));
   };
 
-  const shouldShowFullWidthNativePreview = useIsSmallScreen();
+  const shouldShowFullWidthNativePreview = useMedia("(max-width: 80em)");
   const transformStyle = isOpen ? "translateY(0)" : "translateY(-100%)";
 
   const Handle = forwardRef<
