diff --git a/frontend/src/metabase/querying/expressions/string.ts b/frontend/src/metabase/querying/expressions/string.ts
index 5d2497a0c01..3d4b82e7b6d 100644
--- a/frontend/src/metabase/querying/expressions/string.ts
+++ b/frontend/src/metabase/querying/expressions/string.ts
@@ -104,7 +104,7 @@ export function unquoteString(
       } else if (ch === OPEN || ch === CLOSE) {
         str += ch;
       } else {
-        str += BACKSLASH + ch;
+        str += ch;
       }
     } else if (ch === CLOSE) {
       // skip last delimiter
