--- a/src/metabase/query_processor/middleware/results_metadata.clj
+++ b/src/metabase/query_processor/middleware/results_metadata.clj
@@ -142,12 +142,8 @@
   "Post-processing middleware that records metadata about the columns returned when running the query. Returns an rff."
   [{{:keys [skip-results-metadata?]} :middleware, :as query} :- ::qp.schema/any-query
    rff                                                       :- ::qp.schema/rff]
-  (if (or skip-results-metadata?
-          (qp.settings/ludicrous-speed))
-    rff
-    (let [record! (partial record-metadata! query)]
-      (fn record-and-return-metadata!-rff* [metadata]
-        (insights-xform metadata record! (rff metadata))))))
+  ;; insights-xform unconditionally skipped — all queries bypass fingerprinting
+  rff)

 (mu/defn store-previous-result-metadata!
   "Store the previous value of a card's result metadata in the qp.store"
