--- a/src/metabase/dashboards_rest/api.clj
+++ b/src/metabase/dashboards_rest/api.clj
@@ -1408,7 +1408,8 @@
                body
                {:dashboard-id dashboard-id
                 :card-id      card-id
-                :dashcard-id  dashcard-id}))))
+                :dashcard-id  dashcard-id
+                :middleware   {:skip-results-metadata? true}}))))

 ;; TODO (Cam 2025-11-25) please add a response schema to this API endpoint, it makes it easier for our customers to
 ;; use our API + we will need it when we make auto-TypeScript-signature generation happen
