diff --git a/frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts b/frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts
index c95f68b6ed4..4631dba579c 100644
--- a/frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts
+++ b/frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts
@@ -19,7 +19,9 @@ import { getWaterfallChartModel } from "metabase/visualizations/echarts/cartesia
 import { getWaterfallChartOption } from "metabase/visualizations/echarts/cartesian/waterfall/option";
 import { useBrowserRenderingContext } from "metabase/visualizations/hooks/use-browser-rendering-context";
 import type { VisualizationProps } from "metabase/visualizations/types";
-import type { CardDisplayType } from "metabase-types/api";
+import type { CardDisplayType, TimelineEventId } from "metabase-types/api";
+
+const NO_SELECTED_TIMELINE_EVENT_IDS: TimelineEventId[] = [];
 
 export function useModelsAndOption(
   {
@@ -31,9 +33,8 @@ export function useModelsAndOption(
     height,
     hiddenSeries = new Set(),
     timelineEvents,
-    selectedTimelineEventIds,
+    selectedTimelineEventIds = NO_SELECTED_TIMELINE_EVENT_IDS,
     onRender,
-    hovered,
     isFullscreen,
     gridSize,
   }: VisualizationProps,
@@ -86,11 +87,9 @@ export function useModelsAndOption(
       gridSize,
     );
 
-    if (model.dimensionModel.column) {
-      model.dimensionModel.column.display_name = tc(
-        model.dimensionModel.column.display_name,
-      );
-    }
+    model.dimensionModel.column.display_name = tc(
+      model.dimensionModel.column.display_name,
+    );
     return model;
   }, [
     card.display,
@@ -117,29 +116,10 @@ export function useModelsAndOption(
   );
 
   const timelineEventsModel = useMemo(
-    () =>
-      getTimelineEventsModel(
-        chartModel,
-        chartLayout,
-        timelineEvents ?? [],
-        renderingContext,
-      ),
-    [chartModel, chartLayout, timelineEvents, renderingContext],
+    () => getTimelineEventsModel(chartModel, chartLayout, timelineEvents ?? []),
+    [chartModel, chartLayout, timelineEvents],
   );
 
-  const selectedOrHoveredTimelineEventIds = useMemo(() => {
-    const ids = [];
-
-    if (selectedTimelineEventIds != null) {
-      ids.push(...selectedTimelineEventIds);
-    }
-    if (hovered?.timelineEvents != null) {
-      ids.push(...hovered.timelineEvents.map((e) => e.id));
-    }
-
-    return ids;
-  }, [selectedTimelineEventIds, hovered?.timelineEvents]);
-
   const tooltipOption = useMemo(() => {
     return getTooltipOption(
       chartModel,
@@ -164,8 +144,9 @@ export function useModelsAndOption(
           chartModel as WaterfallChartModel,
           width,
           chartLayout,
+          hasTimelineEvents,
           timelineEventsModel,
-          selectedOrHoveredTimelineEventIds,
+          selectedTimelineEventIds,
           settings,
           shouldAnimate,
           renderingContext,
@@ -175,8 +156,9 @@ export function useModelsAndOption(
         baseOption = getScatterPlotOption(
           chartModel as ScatterPlotModel,
           chartLayout,
+          hasTimelineEvents,
           timelineEventsModel,
-          selectedOrHoveredTimelineEventIds,
+          selectedTimelineEventIds,
           settings,
           width,
           shouldAnimate,
@@ -187,8 +169,9 @@ export function useModelsAndOption(
         baseOption = getCartesianChartOption(
           chartModel as CartesianChartModel,
           chartLayout,
+          hasTimelineEvents,
           timelineEventsModel,
-          selectedOrHoveredTimelineEventIds,
+          selectedTimelineEventIds,
           settings,
           width,
           shouldAnimate,
@@ -207,11 +190,18 @@ export function useModelsAndOption(
     tooltipOption,
     chartModel,
     chartLayout,
+    hasTimelineEvents,
     timelineEventsModel,
-    selectedOrHoveredTimelineEventIds,
+    selectedTimelineEventIds,
     settings,
     renderingContext,
   ]);
 
-  return { chartModel, timelineEventsModel, option, renderingContext };
+  return {
+    chartModel,
+    chartLayout,
+    timelineEventsModel,
+    option,
+    renderingContext,
+  };
 }
