diff --git a/frontend/src/metabase/query_builder/store/selectors.ts b/frontend/src/metabase/query_builder/store/selectors.ts
index 57379a3f965..5ceadddaf06 100644
--- a/frontend/src/metabase/query_builder/store/selectors.ts
+++ b/frontend/src/metabase/query_builder/store/selectors.ts
@@ -1048,9 +1048,7 @@ export const getIsVisualized = createSelector(
     ((question.display() !== "table" &&
       question.display() !== "pivot" &&
       question.display() !== "list") ||
-      (settings != null &&
-        (settings["table.pivot"] ||
-          (question.display() === "table" && settings["table.pivot_column"])))), // last case - pivot_column is set but display is set to table viz (#56094)
+      (settings != null && settings["table.pivot"])),
 );
 
 export const getIsLiveResizable = createSelector(
