diff --git a/frontend/src/metabase/query_builder/store/selectors.ts b/frontend/src/metabase/query_builder/store/selectors.ts
index 57379a3f965..5379dc6bd22 100644
--- a/frontend/src/metabase/query_builder/store/selectors.ts
+++ b/frontend/src/metabase/query_builder/store/selectors.ts
@@ -1046,8 +1046,7 @@ export const getIsVisualized = createSelector(
     question &&
     // table is the default
     ((question.display() !== "table" &&
-      question.display() !== "pivot" &&
-      question.display() !== "list") ||
+      question.display() !== "pivot") ||
       (settings != null &&
         (settings["table.pivot"] ||
           (question.display() === "table" && settings["table.pivot_column"])))), // last case - pivot_column is set but display is set to table viz (#56094)
