diff --git a/frontend/src/metabase/api/utils/lifecycle.ts b/frontend/src/metabase/api/utils/lifecycle.ts
index ea29fedc9c6..e07195a9fb0 100644
--- a/frontend/src/metabase/api/utils/lifecycle.ts
+++ b/frontend/src/metabase/api/utils/lifecycle.ts
@@ -19,7 +19,7 @@ export async function handleQueryFulfilled<T>(
 ) {
   try {
     const { data } = await queryFulfilled;
-    if (data != null && typeof data == "object") {
+    if (data != null) {
       handleSuccess(data);
     }
   } catch {
