diff --git a/src/metabase/lib/native.cljc b/src/metabase/lib/native.cljc
index a5fc78c21e2..872276706ed 100644
--- a/src/metabase/lib/native.cljc
+++ b/src/metabase/lib/native.cljc
@@ -217,10 +217,8 @@
   [query :- ::lib.schema/query
    metadata-provider :- ::lib.schema.metadata/metadata-providerable]
   (assert-native-stage (lib.util/query-stage query 0))
-  (let [stages-without-fields (->> (:stages query)
-                                   (mapv (fn [stage]
-                                           (update stage :template-tags update-vals #(dissoc % :dimension)))))]
-    (lib.query/query-with-stages metadata-provider stages-without-fields)))
+  ;; Changing the database should also clean up template tags, see #31926
+  (lib.query/query-with-stages metadata-provider (:stages query)))
 
 (mu/defn native-extras :- [:maybe ::native-extras]
   "Returns the extra keys for native queries associated with this query."
