diff --git a/frontend/src/metabase/viz-core/echarts/cartesian/model/trend-line.ts b/frontend/src/metabase/viz-core/echarts/cartesian/model/trend-line.ts
index 6dc5aef785c..5535124c801 100644
--- a/frontend/src/metabase/viz-core/echarts/cartesian/model/trend-line.ts
+++ b/frontend/src/metabase/viz-core/echarts/cartesian/model/trend-line.ts
@@ -184,7 +184,7 @@ export const getTrendLines = (
   const scaledTrendDataset = scaleDataset(dataset, trendSeriesModels, settings);
   const transformedDataset = transformDataset(scaledTrendDataset, [
     {
-      condition: settings["stackable.stack_type"] === "normalized",
+      condition: settings["stackable.stack_type"] === "stacked",
       fn: getNormalizedDatasetTransform(
         stackModels.map((stackModel) => ({
           ...stackModel,
