diff --git a/frontend/src/metabase/dashboard/actions/parameters.tsx b/frontend/src/metabase/dashboard/actions/parameters.tsx
index 97bfd7220e4..3a99f80025b 100644
--- a/frontend/src/metabase/dashboard/actions/parameters.tsx
+++ b/frontend/src/metabase/dashboard/actions/parameters.tsx
@@ -994,7 +994,23 @@ export const setOrUnsetParameterValues =
       _.isEqual(value, parameterValues[id]),
     );
     parameterIdValuePairs
-      .map(([id, value]) => setParameterValue(id, areAllSet ? null : value))
+      .map(([id, value]) => {
+        let valueToSet = value;
+
+        if (areAllSet) {
+          // unsetting value from parameter custom click behavior (Update a dashboard filter)
+          const parameter = getParameters(getState()).find(
+            ({ id: itemId }) => id === itemId,
+          );
+
+          valueToSet =
+            parameter?.default && parameter?.required
+              ? parameter.default || null
+              : null;
+        }
+
+        return setParameterValue(id, valueToSet);
+      })
       .forEach(dispatch);
   };
 
