diff --git a/frontend/src/metabase/nav/components/search/SearchBar/SearchBar.tsx b/frontend/src/metabase/nav/components/search/SearchBar/SearchBar.tsx
index 92b366f66f1..1edd45a0458 100644
--- a/frontend/src/metabase/nav/components/search/SearchBar/SearchBar.tsx
+++ b/frontend/src/metabase/nav/components/search/SearchBar/SearchBar.tsx
@@ -145,7 +145,7 @@ function SearchBar({
   }, [previousLocation, location]);
 
   useEffect(() => {
-    if (previousLocation !== location && navigationType !== "REPLACE") {
+    if (previousLocation?.pathname !== location.pathname) {
       // deactivate search when page changes
       setInactive();
     }
