diff --git a/src/metabase/users/settings.clj b/src/metabase/users/settings.clj
index b187e848c3f..4e87589b5c5 100644
--- a/src/metabase/users/settings.clj
+++ b/src/metabase/users/settings.clj
@@ -16,10 +16,7 @@
   (deferred-tru "The last database a user has selected for a native query or a native model.")
   :user-local :only
   :visibility :authenticated
-  :type :integer
-  :getter (fn []
-            (when-let [id (setting/get-value-of-type :integer :last-used-native-database-id)]
-              (when (t2/exists? :model/Database :id id) id))))
+  :type :integer)
 
 (defsetting dismissed-excel-pivot-exports-banner
   (deferred-tru "Toggle which is true after a user has dismissed the excel pivot exports banner.")
