diff --git a/src/metabase/lib/schema/parameter.cljc b/src/metabase/lib/schema/parameter.cljc
index d36cf9af2e8..93645b4835f 100644
--- a/src/metabase/lib/schema/parameter.cljc
+++ b/src/metabase/lib/schema/parameter.cljc
@@ -342,8 +342,8 @@
         :number/between
         (let [[l u] (:value param)]
           (cond-> param
-            (nil? u) (assoc :type :number/>=, :value [l])
-            (nil? l) (assoc :type :number/<=, :value [u])))
+            (and l (nil? u)) (assoc :type :number/>=, :value [l])
+            (and u (nil? l)) (assoc :type :number/<=, :value [u])))
         param))))
 
 (mr/def ::id
