diff --git a/frontend/src/metabase/actions/hooks/use-action-initial-values.ts b/frontend/src/metabase/actions/hooks/use-action-initial-values.ts
index 59cdad8b403..45f9cf54a9e 100644
--- a/frontend/src/metabase/actions/hooks/use-action-initial-values.ts
+++ b/frontend/src/metabase/actions/hooks/use-action-initial-values.ts
@@ -29,10 +29,10 @@ export const useActionInitialValues = ({
   const hasPrefetchedValues = Object.keys(prefetchedInitialValues).length > 0;
 
   useEffect(() => {
-    if (shouldPrefetch && !hasPrefetchedValues) {
+    if (shouldPrefetch) {
       prefetchValues();
     }
-  }, [shouldPrefetch, hasPrefetchedValues, prefetchValues]);
+  }, [shouldPrefetch, prefetchValues]);
 
   return {
     error,
