diff --git a/frontend/src/metabase/parameters/utils/parameter-parsing.ts b/frontend/src/metabase/parameters/utils/parameter-parsing.ts
index 7f0409c4b76..049410ab6d2 100644
--- a/frontend/src/metabase/parameters/utils/parameter-parsing.ts
+++ b/frontend/src/metabase/parameters/utils/parameter-parsing.ts
@@ -27,20 +27,14 @@ export function getParameterValueFromQueryParams(
   const lastUsedValues: ParameterValuesMap = lastUsedParametersValues || {};
 
   const maybeParameterValue = params[parameter.slug || parameter.id];
-  const hasQueryParams = Object.keys(params).length > 0;
 
   // don't use the default with "param=" because it indicates an unset/cleared parameter value
   if (maybeParameterValue === "") {
     return null;
   } else if (maybeParameterValue == null) {
-    // If there is a parameter with a value set in the URL, do not use last used
-    // parameter values (metabase#48524). This avoids a case where some
-    // parameters are set via the URL and some have values from the last run.
-    if (hasQueryParams) {
-      return parameter.default ?? null;
-    } else {
-      return lastUsedValues[parameter.id] ?? parameter.default ?? null;
-    }
+    // first try to use last used parameter value then try to use the default if
+    // the parameter is not present in the query params
+    return lastUsedValues[parameter.id] ?? parameter.default ?? null;
   }
 
   const parsedValue = parseParameterValue(maybeParameterValue, parameter);
