diff --git a/frontend/src/metabase/query_builder/store/reducer.ts b/frontend/src/metabase/query_builder/store/reducer.ts
index 3bd08dd101e..2ec80ae3b9b 100644
--- a/frontend/src/metabase/query_builder/store/reducer.ts
+++ b/frontend/src/metabase/query_builder/store/reducer.ts
@@ -298,7 +298,8 @@ const uiControls = createReducer<QueryBuilderUIControls>(
       .addCase(OPEN_QUESTION_INFO, (state) =>
         setUIControls(state, {
           ...UI_CONTROLS_SIDEBAR_DEFAULTS,
-          ...CLOSED_NATIVE_EDITOR_SIDEBARS,
+          isShowingSnippetSidebar: false,
+          isShowingDataReference: false,
           isShowingQuestionInfoSidebar: true,
         }),
       )
