diff --git a/frontend/src/metabase/dashboard/hooks/use-is-parameter-panel-sticky.ts b/frontend/src/metabase/dashboard/hooks/use-is-parameter-panel-sticky.ts
index 4df506c82e8..7dccd27ea83 100644
--- a/frontend/src/metabase/dashboard/hooks/use-is-parameter-panel-sticky.ts
+++ b/frontend/src/metabase/dashboard/hooks/use-is-parameter-panel-sticky.ts
@@ -26,10 +26,11 @@ export function useIsParameterPanelSticky({
     sentinel.style.top = "0";
     sentinel.style.visibility = "hidden";
 
-    // Insert sentinel before the parent of our sticky element
-    const parent = parameterPanelRef.current.parentElement;
-    if (parent) {
-      parent.insertBefore(sentinel, parameterPanelRef.current);
+    if (parameterPanelRef.current) {
+      parameterPanelRef.current.insertBefore(
+        sentinel,
+        parameterPanelRef.current.firstChild,
+      );
     }
 
     const settings: IntersectionObserverInit = {
diff --git a/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx b/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx
index d95b0c3569a..2faf669ee34 100644
--- a/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx
+++ b/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx
@@ -253,7 +253,7 @@ export const EmbedFrame = ({
         {/* show floating header buttons if there is no title */}
         {headerButtons && !titled ? headerButtons : null}
 
-        <span ref={parameterPanelRef} />
+        <span ref={parameterPanelRef} style={{ position: "relative" }} />
         {hasVisibleParameters && (
           <FullWidthContainer
             className={cx(EmbedFrameS.ParameterPanel, {
