diff --git a/src/metabase/lib/filter/update.cljc b/src/metabase/lib/filter/update.cljc
index 95095b5bf68..7a967a80066 100644
--- a/src/metabase/lib/filter/update.cljc
+++ b/src/metabase/lib/filter/update.cljc
@@ -235,13 +235,8 @@
     longitude-column                             :- :some
     {:keys [north east south west], :as _bounds} :- [:ref ::lat-lon.bounds]]
    (-> query
-       (remove-existing-filters-against-column-checking-subclauses stage-number latitude-column)
-       (remove-existing-filters-against-column-checking-subclauses stage-number longitude-column)
-       (lib.filter/filter stage-number
-                          (if (<= west east)
-                            ;; bounds do not cross the antimerdian. A single :inside filter suffices.
-                            (lib.filter/inside latitude-column longitude-column north west south east)
-                            ;; bounds do cross the antimerdian. Split into two filters for the east and west sides.
-                            (lib.filter/or
-                             (lib.filter/inside latitude-column longitude-column north west south 180)
-                             (lib.filter/inside latitude-column longitude-column north -180 south east)))))))
+       (remove-existing-filters-against-column stage-number latitude-column)
+       (remove-existing-filters-against-column stage-number longitude-column)
+       (lib.filter/filter stage-number (let [[lat-min lat-max] (sort [north south])
+                                             [lon-min lon-max] (sort [east west])]
+                                         (lib.filter/inside latitude-column longitude-column lat-max lon-min lat-min lon-max))))))
